File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var distinctAverages = function(nums) {
        let set = new Set();
        nums.sort((a,b)=>{return a - b});
        let l = 0;
        let r = nums.length - 1;
        while(l <= r){
            let avg = (nums[l] + nums[r])/2;
            set.add(avg);
            l++;
            r--;
        }
        return set.size;
};

File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.java
class Solution {
    public int distinctAverages(int[] nums) {
        HashSet<Double> set = new HashSet<Double>();
        Arrays.sort(nums);
        int l = 0;
        int r = nums.length - 1;
        while(l <= r){
            double avg = (nums[l] + nums[r])/2.0;
            set.add(avg);
            l++;
            r--;
        }
        return set.size();
    }
}